### DISCUSSION TITLE
Peering Manager for more router platforms

### DISCUSSION THREAD

**Discussion by wtremmel** [peering-manager/peering-manager#634] [Ideas]
Hi,

currently Peering Manager supports router platforms Cisco, Juniper and everything else supported by [Napalm](https://napalm.readthedocs.io/en/latest/support/index.html) 

What I find missing are "cheaper" routers like [Mikrotik](https://mikrotik.com), which are extremely popular in [some regions](https://www.de-cix.in). 

How to support them? Can it be done in any way? Sponsor a Napalm developer? Or add Mikrotik support directly into Peering Manager?

Are there other platforms which are missing ?

best regards
Wolfgang

**Comment by boggits** [comment]
There is a napalm extension (with flaws) for Mikrotik

https://github.com/napalm-automation-community/napalm-ros

I'd suggest that going via the Napalm route give the broadest reach

**Reply by gmazoyer** [reply]
As @boggits mentioned, PM loads the appropriate driver based on the platform definition, that you can add in [here](https://demo.peering-manager.net/devices/platforms/) for instance. You'll need to fill the `NAPALM driver` field with the string identifier of the driver.

The NAPALM driver will need to be installed in the same environment (virtual or not) that PM is running in. If the driver is successfully loaded then PM should be able to use some of its functions. However, be aware that some NAPALM drivers do not implement all the features PM needs.

**Reply by jbemmel** [reply]
I have a PR ready to add support for Nokia SR OS, it would add a crypto driver to handle password hashing and a dependency on the NAPALM driver (napalm-sros) such that it gets installed in the same environment.

Would that be acceptable?

**Reply by gmazoyer** [reply]
I believe that could be acceptable as long as the NAPALM driver stays maintained.
